### DISCUSSION TITLE
Stopping a Pattern

### DISCUSSION THREAD

**Discussion by macduff111** [supriya-project/supriya#356] [Q&A] [answered]
Hi,
I can now send patterns to synthdefs, and am excited that I can control scsynth from python! I never got on with sclang, but scsynth is v powerful with a great architecture.
Anyway, my question - how do I stop a Pattern once it has been started with play() ?
Thanks

**Comment by josiah-wolf-oberholtzer** [comment, marked as answer]
Save a reference to the pattern player and call `.stop()` on it:

```
player = my_pattern.play(...)
player.stop()
```

Patterns themselves are just stateless iterables. When you call `my_pattern.play()` you're creating a `PatternPlayer` object which maintains the state of iteration through the pattern, and coordinates between that iterator, the clock, and the synthesis execution context (typically a `Server`).

**Comment by josiah-wolf-oberholtzer** [comment]
Also, nit about verbiage. SynthDefs don't receive anything. Think of them like classes or templates. When you issue a `/s_new` command under the hood, you're allocating an instance of the SynthDef. That occurs e.g. when an `EventPattern`'s `NoteEvent` gets performed against the execution context by the pattern player.

**Reply by macduff111** [reply]
Yes, understood, my terminology a bit loose. "I can select which SynthDef a Pattern binds to" would be better way of putting it perhaps.
